crates/core/tedge_agent/src/agent.rs

```diff
@@ -179,8 +179,8 @@ impl Agent {
         runtime.spawn(restart_actor_builder).await?;
         runtime.spawn(software_update_builder).await?;
         runtime.spawn(converter_actor_builder).await?;
-        runtime.spawn(health_actor).await?;
         runtime.spawn(tedge_to_te_converter).await?;
+        runtime.spawn(health_actor).await?;
```

Comment:
Start all the other actors before the health actor starts.

Reply:
Why? What was the issue? How this fix improves things?

Reply:
There was a suspicion that some of the integration test failures where potentially caused by the converter actor not being "ready" before the messages were received by the broker (as the test would check the health status of the `tedge-agent` before publishing to the measurements topic.

Reply:
Yes, as @reubenmiller mentioned it's to make sure the `tedge-agent` is up. This is not completely fail-safe as the `health actor` does not check whether all the other actors are started successfully or not. But, at least we are sure that the `te converter` actor spawned (though it might hang and not be completely ready to serve the requests) before starting the `health` actor.

Reply:
Okay. So you suspect that:

1. The `health_actor` sends an `up` status before `tedge_to_te_converter` is spawned.
2. The test sends a measurement on `tedge` and expects the same on `te`.
3. The test gives up waiting for the message on `te` before `tedge_to_te_converter` got a chance to be spawned and notified of the measurement on `tedge`.

I see that as extremely unlikely unless `tedge_to_te_converter` hangs .i.e. is buggy. Indeed, the difference between the time scale of tokio context-switch and system test timeouts is of an order of magnitude 10^6 .
 
I have nothing against this change. However, if you observed such failures then the issue might be deeper.